src/runtime/typemanager.cs

```diff
@@ -680,6 +884,79 @@ internal static void InitializeNativeCodePage()
         }
         #endregion
 
+        /// <summary>
+        /// Given a managed Type that provides the implementation for the type,
+        /// create a PY_TYPE_SLOT array to be used for PyType_FromSpec.
+        /// </summary>
+        internal static PY_TYPE_SLOT[] CreateSlotArray(Type impl)
+        {
+            // We work from the most-derived class up; make sure to get
+            // the most-derived slot and not to override it with a base
+            // class's slot.
+            var seen = new HashSet<string>();
+            var typeslots = new List<PY_TYPE_SLOT>();
+
+            while (impl != null)
+            {
+                MethodInfo[] methods = impl.GetMethods(tbFlags);
+                foreach (MethodInfo method in methods)
+                {
+                    string name = method.Name;
+                    if (!(name.StartsWith("tp_") ||
+                          name.StartsWith("nb_") ||
+                          name.StartsWith("sq_") ||
+                          name.StartsWith("mp_") ||
+                          name.StartsWith("bf_")
+                    ))
+                    {
+                        continue;
+                    }
+
+                    if (seen.Contains(name))
+                    {
+                        continue;
+                    }
+
+                    typeslots.Add(InitializeSlot(TypeSlots.getSlotNumber(name), method));
+                    seen.Add(name);
```

Comment:
I vaguely remember seeing this code elsewhere. You should remove it from the original location.

Reply:
@lostmsu its in this same file.  I'll cleanup once I get things working